### ISSUE TITLE
Unable to run this demo

### ISSUE THREAD

**Issue by aakashysharma** [reanahub/reana-demo-cms-h4l#26]
I've setup a new installation of REANA and I've been trying to run all the demo projects available. So, far this demo and cms-reco fails. The others demos run fine. I've attached the logs file I collected from reana-client. 
[logs.txt](https://github.com/reanahub/reana-demo-cms-h4l/files/13390657/logs.txt)
Any pointers will be helpful.

**Comment by tiborsimko** [comment]
In the logs you posted I'm seeing:

```
2023-11-17 10:30:47,400 | reana-workflow-engine-snakemake | Thread-1 | INFO | scram job is failed. 
```

So the problem is already with the first step of the workflow.

If you have set up your own cluster, you could do `kubectl get pods` and see if there are any failed `reana-run-job-...` pods and then you could inspect them for getting more information about the source of the error.

However, before doing that, just to double check whether this may not be something similar as in https://github.com/reanahub/reana-demo-bsm-search/issues/31, can you please check whether the error also happens for your installation when you do:

```console
$ git clone --depth=1 https://github.com/reanahub/reana-demo-cms-h4l
$ cd reana-demo-cms-h4l
$ reana-client run -w h4l
```

**Comment by aakashysharma** [comment]
It fails with the suggested steps as well. I've attached the logs. 
[depth1logs.txt](https://github.com/reanahub/reana-demo-cms-h4l/files/13390936/depth1logs.txt)

**Comment by tiborsimko** [comment]
The attached file shows that you have been triggering the Serial worklfow engine in the second run, not the Snakemake workflow engine as in your first run. This indicates that there is probably something wrong with your `reana.yaml`.  Can you start the cloning in a fully empty directory, say `/tmp/`,  just to make fully sure about the files?

I'll also update the `README.rst` to show the expected `reana.yaml` content explicitly.

**Comment by aakashysharma** [comment]
depth1logs.txt shows the errors if I use `reana-client run -w h4l`. If I follow the step-by-step guide on the GitHub page, it gives similar error as earlier. 
[failed.txt](https://github.com/reanahub/reana-demo-cms-h4l/files/13390998/failed.txt)

**Comment by aakashysharma** [comment]
I cleaned up and started from the beginning to clone the repo.

**Comment by tiborsimko** [comment]
@aakashysharma Is your REANA cluster deployed on ARM processors, such as on Apple silicon? I know that older CMSSW container images (as used in this very example) won't run on Apple silicon natively. The example should run well on x86_64 (AMD, Intel) processors though; I just rechecked it.

**Comment by aakashysharma** [comment]
Nope, old classical x86_64 system. `Intel(R) Xeon(R) Gold 6238R CPU @ 2.20GHz` with Debian 11.

**Comment by tiborsimko** [comment]
That should work then.  FYI this is what I'm getting running this example:

```console
$ reana-client status -w h4l
~                                                                                                                              │NAME   RUN_NUMBER   CREATED               STARTED               ENDED                 STATUS     PROGRESS
~                                                                                                                              │h4l    5            2023-11-17T11:06:48   2023-11-17T11:07:15   2023-11-17T11:09:07   finished   4/4
```

Some things to explore would be whether you have enough memory available in the cluster.  The container image is quite large (9.64 GB).

You can look at `kubectl get events` in the namespace where REANA was deployed in order to see whether there isn't anything interesting logged there for the given workflow at hand.  Also, as mentioned previously, please check `kubectl get pods | grep reana-run` in the namespace where REANA is installed, to see whether some of the pods remained there in `Error` state; inspecting those would give more information then what is seen by `reana-client logs`. For example, in your `failed.txt` file, the scramming job was reported as being `reana-run-job-df49b3c2-6563-452b-9535-9f69cbc7f5f0`, so you could look for that ID amongst Kubernetes jobs and pods and events.

**Comment by aakashysharma** [comment]
I'm still investigating this issue. The system has enough memory (~256GB) available for the pods. We wanted to extend this deployment by adding some Nvidia Jetson nodes. But your recent comment about ARM made me thinking. Is REANA not supported on ARM-based CPUs? Or is it just this example code that can run into problems with ARM-based systems. Thanks for your help.

**Comment by tiborsimko** [comment]
The REANA master branch cluster runs well on arm64 hardware such as Apple M2. The latest stable release (0.9.1) is publishing amd64 images only, however we are preparing a bug fix release (0.9.2) that will be the first one to contain multi-platform images (amd64, arm64). So there should be no problem for you on this side for your tests; if you switch to using ARM-based system, the REANA cluster will run using ARM-optimised images. 

The problem is more the latter: this particular CMS-specific example is using a particular container image that needs the amd64 platform for running...
